generator/internal/service_code_generator.cc

```diff
@@ -156,6 +157,13 @@ bool ServiceCodeGenerator::HasBidirStreamingMethod() const {
                      });
 }
 
+bool ServiceCodeGenerator::HasExplicitRoutingMethod() const {
+  return std::any_of(methods_.begin(), methods_.end(),
+                     [](google::protobuf::MethodDescriptor const& m) {
+                       return m.options().HasExtension(google::api::routing);
+                     });
+}
```

Comment:
Q.  Does this deserve a "predicate_utils" delegate on the `MethodDescriptor`, like the others here?